server/routing/routing_remote.go

```diff
@@ -168,7 +171,30 @@ func newRemote(ctx context.Context,
 
 	routeAPI.service = rpcService
 
-	routeAPI.cleanupManager = NewCleanupManager(dstore, storeAPI, server)
+	// Initialize GossipSub manager if enabled
+	// Protocol parameters (topic, message size) are defined in pubsub.constants
+	// and are NOT configurable to ensure network-wide compatibility
+	if opts.Config().Routing.GossipSub.Enabled {
+		pubsubManager, err := pubsub.New(ctx, server.Host())
```

Comment:
the pubsubManager is never closed on success (somewhere in the routing remote) which can result in potential memory leak and network-hogging